File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        char temp = '0';
        int end = s.length -1;
        for (int i = 0; i < s.length / 2; i++) {
            temp = s[i];
            s[i] = s[end - i];
            s[end - i] = temp;
        }
    }
}

File: 0344-reverse-string/0344-reverse-string.kt
class Solution {
    fun reverseString(s: CharArray): Unit {
        var start = 0
        var end = s.size -1

        while (start < end) {
        // also를 이용해 우아하게 스왑
        s[start] = s[end].also {s[end] = s[start]}

        // 앞쪽 문자는 한 칸 뒤로, 뒤쪽 문자는 한 칸 앞으로 이동
        start++
        end--
        }
    }
}